### PR TITLE
PLAT-108973: [Agate][GMRSI] Supporting input cancel button

### PR DESCRIPTION
**Pull request by adrian-cocoara-lgp**
Enact-DCO-1.0-Signed-off-by: Adrian Cocoara adrian.cocoara@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Input needs a clear button. When clicked, it should set the value of input to an empty string

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Implemented a `clearInputButton` for input
Updated unit, UI and screenshot tests with the new feature

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
N/A

### Links
[//]: # (Related issues, references)
PLAT-108973

### CODE REVIEW

File: Input/Input.module.less [hunk 1]
```diff
@@ -5,6 +5,8 @@
 @import "../styles/variables.less";
 
 .decorator {
+	align-items: center;
```
Comment (daniel-stoian-lgp) [inline Input/Input.module.less:8]: For the next reviewer: this was needed to solve vertical alignment of text and icons on some skins: 
image

File: styles/variables-base.less [hunk 2]
```diff
@@ -382,8 +382,8 @@
 @agate-input-small-font-size: 24px;
 @agate-input-small-height: (@agate-button-small-height - (@agate-input-decorator-border-width * 2));
 @agate-input-small-icon-after-padding: 0 9px 0 9px;
-@agate-input-small-icon-align-self: center;
 @agate-input-small-icon-before-padding: @agate-input-small-icon-after-padding;
+@agate-input-small-icon-align-self: center;
```
Comment (daniel-stoian-lgp) [inline styles/variables-base.less:386]: please move it to the same position is was before :)

**Review by daniel-stoian-lgp — APPROVED** [review]

**Review by 0x64 — APPROVED** [review]
Except for one missing jsdoc comment, it looks good to me!

**Comment by 0x64** [comment]
Please resolve a conflict of CHANGELOG.md.

File: Input/Input.js [hunk 3]
```diff
@@ -41,6 +42,23 @@ const InputBase = kind({
 	name: 'Input',
 
 	propTypes: /** @lends agate/Input.InputBase.prototype */ {
+		/**
+		 * Shows the clear input button.
+		 *
+		 * @type {Boolean}
+		 * @default false
```
Comment (MikyungKim) [inline Input/Input.js:49]: You can remove the default value if it's boolean and has false as default.

File: Input/Input.js [hunk 4]
```diff
@@ -226,6 +244,8 @@ const InputBase = kind({
 	},
 
 	defaultProps: {
+		clearInputButton: false,
```
Comment (MikyungKim) [inline Input/Input.js:247]: You could remove this line too.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**